### PR TITLE
[AMSDK-10268]Handle Analytics response identity events

### PR DESCRIPTION
**Pull request by cacheung**
Listen for Analytics response identity event
Dispatch Avid sync event
Add unit tests

## Description

-Handle Analytics response identity events

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by nporter-adbe — CHANGES_REQUESTED** [review]
We need to move much of this logic into the IdentityState. If you notice nowhere else in the Identity class do we directly interact with the data store. We should only interact with the data store in classes outside of the Identity extension and rather in the business logic classes.

Identity.swift is mainly reserved for event processing.

For an example look at https://github.com/adobe/aepsdk-core-ios/blob/main/AEPIdentity/Sources/IdentityState.swift#L148

**Comment by cacheung** [comment]
Got it, will fix it soon.

File: AEPIdentity/Sources/IdentityState.swift [hunk 1]
```diff
@@ -252,6 +251,37 @@ class IdentityState {
         lastValidConfig = newConfig
     }
 
+    /// Invoke each time when we receive Analytics Response event
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/IdentityState.swift:254]: typo: Invoked

File: AEPIdentity/Sources/IdentityState.swift [hunk 2]
```diff
@@ -252,6 +251,37 @@ class IdentityState {
...
+    /// - Parameters:
+    ///   - event: the event from Analytics Respsonse
+    ///   - eventDispatcher: a function which when invoked dispatches an `Event` to the `EventHub`
+    func handleAnalyticsResponse(event: Event, eventDispatcher: (Event) -> Void) {
+        guard let aid = event.data?[IdentityConstants.Analytics.ANALYTICS_ID] as? String, !aid.isEmpty else {
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/IdentityState.swift:259]: Add reading the `aid` as we do for other event data inside of `Event+Identity`. Then use `event.aid`.

File: AEPIdentity/Sources/IdentityState.swift [hunk 3]
```diff
@@ -252,6 +251,37 @@ class IdentityState {
...
+                IdentityConstants.EventDataKeys.IS_SYNC_EVENT: true,
+                IdentityConstants.EventDataKeys.AUTHENTICATION_STATE: MobileVisitorAuthenticationState.unknown.rawValue,
+            ]
+
+            let avidEvent = Event(name: IdentityConstants.EventNames.AVID_SYNC_EVENT,
+                                  type: EventType.identity,
+                                  source: EventSource.requestIdentity,
+                                  data: syncData)
+            eventDispatcher(avidEvent)
+            identityProperties.isAidSynced = true
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/IdentityState.swift:279]: Let's update identity properties and save to persistence before we dispatch the event.

**Review by nporter-adbe — COMMENTED** [review]
looks good just some minor comments

File: AEPIdentity/Sources/IdentityState.swift (hunk 1 above)
Comment (cacheung) [reply AEPIdentity/Sources/IdentityState.swift]: updated

File: AEPIdentity/Sources/IdentityState.swift (hunk 2 above)
Comment (cacheung) [reply AEPIdentity/Sources/IdentityState.swift]: updated

File: AEPIdentity/Sources/IdentityState.swift (hunk 3 above)
Comment (cacheung) [reply AEPIdentity/Sources/IdentityState.swift]: updated

**Review by nporter-adbe — APPROVED** [review]

**Pull request merged**